texastribune/donations
# Move quarantined donations process to subscriptions

#### What's this PR do?
Updates the slack action blocks created when a donation is quarantined to feed needed data to slack for subscription and payment intent creation and to drop the creation of temporary opportunities and rdos (depending on the process).

#### Why are we doing this? How does it help us?
This is the last piece before we can consider ourselves moved off a "salesforce first" and onto a "stripe first" workflow for donations.

#### How should this be manually tested?
Though not ideal, because of many moving pieces this should be tested in staging.

* First, make sure the ENABLE_BAD_ACTOR_API var is set to True in staging settings
* Next, give a monthly or yearly donation [here](https://donations-testing.herokuapp.com/donate) of $1
    * Use a test card as found [here](https://stripe.com/docs/testing#cards)
* Look for a message to appear on #tech-test-donations
    * Check the timestamp to ensure this message relates to your donation
    * Check the link at the top of the message labeled "Salesforce" to make sure it connects to your Contact info in salesforce
    * Check the other fields and ensure they look accurate based on the information entered in the form above
    * Lastly, click "Approve" at the bottom of the message
* Check stripe to make sure the donation came through as expected...
    * In our stripe dashboard, make sure you are in "Test Mode" (upper corner)
    * Choose the "Customers" tab and you should see the donation you made, listed by the First Name/Last Name you provided
        * <img width="1110" alt="Screen Shot 2023-08-02 at 4 03 38 PM" src="https://github.com/texastribune/donations/assets/88053677/6c140310-6054-475f-b970-484611618cfd">
    * Clicking through to the customer you added should show a screen listing a Sustaining subscription
        * <img width="1153" alt="Screen Shot 2023-08-02 at 4 09 02 PM" src="https://github.com/texastribune/donations/assets/88053677/b5ee4a5c-6ba4-4a34-baaf-859631c1434a">
* Check salesforce to make sure the correct details were rippled there
    * Login to our salesforce sandbox (deets listed under "Salesforce Test (2023 refactor)")
    * Follow the instructions [here](https://texastribune.atlassian.net/wiki/spaces/TECH/pages/2273837059/Navigating+Salesforce) to ensure that the donation came through on the salesforce end as expected
* Go through the same process as above but with a one-time donation
* Lastly, go through the donation process with either a one-time or recurring donation and reject the donation in #tech-test-donations
    * Checking for a message saying something along the lines of "Donation newly rejected by @username" in that same slack block is really the only checking you can do to make sure this worked

#### How should this change be communicated to end users?
We'll let membership know this has been updated.

#### Are there any smells or added technical debt to note?
I think using slack in this way is a very clever solution for this problem but it can be hard to follow. Might be worth looking into other possible solutions for this need, but that is way outside of the demands of this project.

#### What are the relevant tickets?
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recFtQ7jHLNJ1MJcH?blocks=hide

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***


#### TODOs / next steps:
